### PR TITLE
docs: add words misspell checker

### PR DESCRIPTION
**Pull request by aanm**
Signed-off-by: André Martins <andre@cilium.io>

### CODE REVIEW

**Comment by aanm** [comment]
test-me-please

File: Documentation/Makefile [hunk 1]
```diff
@@ -50,4 +50,9 @@ cmdref:
 %: Makefile check-requirements
 	@$(ECHO_GEN)_build/$@
 	$(SPHINXBUILD) -b spelling -d "$(BUILDDIR)/doctrees" "$(SOURCEDIR)" "$(BUILDDIR)/spelling"
+	@if [ $(shell wc -l < "$(BUILDDIR)/spelling/output.txt") -eq 0 ]; then \
+		echo "ERROR: Found misspelled words in the documentation! If you believe the"; \
+		echo "words are not misspelled, please add them to ./Documentation/speelling_wordlist.txt"; \
```
Comment (raybejjani) [inline Documentation/Makefile]: is `speelling` misspelled on purpose?

File: Documentation/Makefile [hunk 2]
```diff
@@ -50,4 +50,9 @@ cmdref:
 %: Makefile check-requirements
 	@$(ECHO_GEN)_build/$@
 	$(SPHINXBUILD) -b spelling -d "$(BUILDDIR)/doctrees" "$(SOURCEDIR)" "$(BUILDDIR)/spelling"
+	@if [ $(shell wc -l < "$(BUILDDIR)/spelling/output.txt") -eq 0 ]; then \
```
Comment (raybejjani) [inline Documentation/Makefile]: Should this be `-eq` or `-ne`?

**Review by raybejjani — CHANGES_REQUESTED** [review]

**Comment by aanm** [comment]
test-me-please

File: Documentation/Makefile (hunk 1 above)
Comment (aanm) [reply Documentation/Makefile]: fixed

File: Documentation/Makefile (hunk 2 above)
Comment (aanm) [reply Documentation/Makefile]: fixed

**Review by raybejjani — APPROVED** [review]
Should we print the list of misspelled words? or instructions on what to run to get them? Most people will see this output as part of a CI run so it won't be too clear, perhaps.

**Comment by aanm** [comment]
>Should we print the list of misspelled words? or instructions on what to run to get them? Most people will see this output as part of a CI run so it won't be too clear, perhaps.

@raybejjani The list is printed right before the warning. To re-run them we have the `make render-docs` target.

```
checking consistency... done
preparing documents... done
troubleshooting.rst:15:maiddntain:ooting                                                                                                                                                                                                                                                                                     

Spelling checker messages written to /srv/Documentation/_build/spelling/output.txt
WARNING: Found 1 misspelled words
build succeeded, 1 warning.
WARNING: Not copying tabs assets! Not compatible with spelling builder
ERROR: Found misspelled words in the documentation! If you believe the
words are not misspelled, please add them to ./Documentation/spelling_wordlist.txt
```

**Review by ianvernon — APPROVED** [review]

File: Documentation/bpf.rst [hunk 3]
```diff
@@ -515,12 +515,12 @@ example code can then be rewritten more naturally as:
 Mainstream BPF JIT compilers like ``x86_64`` and ``arm64`` support BPF to BPF
 calls today with others following in near future. BPF to BPF call is an
 important performance optimization since it heavily reduces the generated BPF
-code size and therefore becomes friendlier to a CPU's icache.
+code size and therefore becomes friendlier to a CPU's iCache.
```
Comment (borkmann) [inline Documentation/bpf.rst]: Hmm, why `iCache`? Either we should use `icache` or `instruction cache`. `iCache` seems very weird to me.

File: Documentation/bpf.rst [hunk 4]
```diff
@@ -568,7 +568,7 @@ issuing a grep for ``HAVE_EBPF_JIT``:
 JIT compilers speed up execution of the BPF program significantly since they
 reduce the per instruction cost compared to the interpreter. Often instructions
 can be mapped 1:1 with native instructions of the underlying architecture. This
-also reduces the resulting executable image size and is therefore more icache
+also reduces the resulting executable image size and is therefore more iCache
```
Comment (borkmann) [inline Documentation/bpf.rst]: Same here, rest of the fixes look good to me, thanks!

File: Documentation/bpf.rst (hunk 3 above)
Comment (aanm) [reply Documentation/bpf.rst]: I will pick up `instruction cache` since it is more clear.

**Comment by aanm** [comment]
test-me-please

**Review by borkmann — APPROVED** [review]
lgtm

**Comment by aanm** [comment]
test-me-please

**Pull request merged**